Mathlib/Data/Set/Basic.lean

```diff
@@ -2158,6 +2145,53 @@ theorem subset_symmDiff_union_symmDiff_right (h : Disjoint t u) : s ⊆ s ∆ t
   h.le_symmDiff_sup_symmDiff_right
 #align set.subset_symm_diff_union_symm_diff_right Set.subset_symmDiff_union_symmDiff_right
 
+/-! ### Lemmas about pairs -/
+
+
+--Porting note: removed `simp` attribute because `simp` can prove it
+theorem pair_eq_singleton (a : α) : ({a, a} : Set α) = {a} :=
+  union_self _
+#align set.pair_eq_singleton Set.pair_eq_singleton
+
+theorem pair_comm (a b : α) : ({a, b} : Set α) = {b, a} :=
+  union_comm _ _
+#align set.pair_comm Set.pair_comm
+
+-- Porting note: first branch after `constructor` used to be by `tauto!`.
+theorem pair_eq_pair_iff {x y z w : α} :
+    ({x, y} : Set α) = {z, w} ↔ x = z ∧ y = w ∨ x = w ∧ y = z := by
+  simp only [Set.Subset.antisymm_iff, Set.insert_subset_iff, Set.mem_insert_iff,
+    Set.mem_singleton_iff, Set.singleton_subset_iff]
+  constructor
+  · rintro ⟨⟨rfl | rfl, rfl | rfl⟩, ⟨h₁, h₂⟩⟩ <;> simp [h₁, h₂] at * <;> simp [h₁, h₂]
+  · rintro (⟨rfl, rfl⟩ | ⟨rfl, rfl⟩) <;> simp
+#align set.pair_eq_pair_iff Set.pair_eq_pair_iff
+
+theorem pair_diff_left {x y : α} (hne : x ≠ y) : ({x, y} : Set α) \ {x} = {y} := by
+  rw [insert_diff_of_mem _ (by exact rfl : x ∈ {x}), diff_singleton_eq_self (by simpa)]
+
+theorem pair_diff_right {x y : α} (hne : x ≠ y) : ({x, y} : Set α) \ {y} = {x} := by
+  rw [pair_comm, pair_diff_left hne.symm]
+
+@[simp]
+theorem pair_subset_iff {x y : α} {s : Set α} : {x,y} ⊆ s ↔ x ∈ s ∧ y ∈ s := by
```

Comment:
Minor style comment: please insert a space after commas in the pair notation, i.e. `{x, y}`, not `{x,y}`. Same goes for the rest of the file.